packages/grpc-js/src/call-stream.ts

```diff
@@ -245,29 +245,32 @@ export class Http2CallStream extends Duplex implements Call {
       stream.on('response', (headers, flags) => {
         switch (headers[HTTP2_HEADER_STATUS]) {
           // TODO(murgatroid99): handle 100 and 101
-          case '400':
+          case 200:
+            this.mappedStatusCode = Status.OK;
+            break;
+          case 400:
             this.mappedStatusCode = Status.INTERNAL;
             break;
-          case '401':
+          case 401:
             this.mappedStatusCode = Status.UNAUTHENTICATED;
             break;
-          case '403':
+          case 403:
             this.mappedStatusCode = Status.PERMISSION_DENIED;
             break;
-          case '404':
+          case 404:
             this.mappedStatusCode = Status.UNIMPLEMENTED;
             break;
-          case '429':
-          case '502':
-          case '503':
-          case '504':
+          case 429:
+          case 502:
+          case 503:
+          case 504:
             this.mappedStatusCode = Status.UNAVAILABLE;
             break;
           default:
             this.mappedStatusCode = Status.UNKNOWN;
         }
 
-        if (flags & http2.constants.NGHTTP2_FLAG_END_STREAM) {
+        if (flags & http2.constants.NGHTTP2_FLAG_END_STREAM || this.mappedStatusCode !== Status.OK) {
```

Comment:
The problem here is that if the `END_STREAM` flag is not set, then there will still be actual trailers later. I think there might be a better solution here that doesn't involve discarding those trailers.

Reply:
Yeah, I figured. I'm open to suggestions!

Reply:
The simplest solution for now would probably be to just remove that `end` event handler I linked to in the issue. I think that check actually doesn't exist in the other implementation.

Reply:
That doesn't work because the status code set in `mappedStatusCode` only gets handled in `handleTrailers`, which never gets called again.

Reply:
Something worth noting here is that in the case that I'm working with, the problem arrises when grpc is is being served behind nginx with `grpc_pass`.

In this case, nginx serves the 404 error - so I don't think the trailers ever come.

Reply:
After some further debugging it seems that the nginx proxy was incorrectly setup in my case. So, I have updated this PR to remove the changes related to the handing of the trailers - it now just fixes the http status code checking (number vs string)